cert/cert.go

```diff
@@ -36,11 +37,29 @@ type CA struct {
 	AuthKey     string `json:"auth_key" yaml:"auth_key"`
 	AuthKeyFile string `json:"auth_key_file" yaml:"auth_key_file"`
 	File        *File  `json:"file,omitempty" yaml:"file,omitempty"`
+	RootCACert  string `json:"root_ca,omitempty" yaml:"root_ca,omitempty"`
 	pem         []byte
 }
 
 func (ca *CA) getRemoteCert() ([]byte, error) {
-	remote := client.NewServer(ca.Remote)
+	var tlsConfig tls.Config
+	if ca.RootCACert != "" {
+		rootCABytes, err := ioutil.ReadFile(ca.RootCACert)
+		if err != nil {
+			return nil, err
+		}
+
+		rootCaCertPool := x509.NewCertPool()
+		ok := rootCaCertPool.AppendCertsFromPEM(rootCABytes)
+		if !ok {
+			return nil, errors.New("failed to parse rootCA certs")
+		}
+		tlsConfig = tls.Config{
+			RootCAs: rootCaCertPool,
+		}
+	}
+
+	remote := client.NewServerTLS(ca.Remote, &tlsConfig)
```

Comment:
Unless I'm missing something, this code change will result in an empty/zero initialized struct being passed for tlsConfig- rather than nil; does the consuming code make a difference in that regard?

More specifically, I want to confirm that these changes won't break the existing pathway, and tests don't exist so it's hard to evaluate this as a drive by reviewer.